dlt/cli/deploy_command.py

```diff
@@ -33,188 +34,367 @@
 REQUIREMENTS_GITHUB_ACTION = "requirements_github_action.txt"
 GITHUB_URL = "https://github.com/"
 DLT_DEPLOY_DOCS_URL = "https://dlthub.com/docs/walkthroughs/deploy-a-pipeline"
+DLT_AIRFLOW_GCP_DOCS_URL = "https://dlthub.com/docs/running-in-production/orchestrators/airflow-gcp-cloud-composer"
+AIRFLOW_GETTING_STARTED = "https://airflow.apache.org/docs/apache-airflow/stable/start.html"
+AIRFLOW_DAG_TEMPLATE_SCRIPT = "dag_template.py"
+AIRFLOW_CLOUDBUILD_YAML = "cloudbuild.yaml"
 
 
-def deploy_command(pipeline_script_path: str, deployment_method: str, schedule: str, run_on_push: bool, run_on_dispatch: bool, branch: str = None) -> None:
+class DeploymentMethods(Enum):
+    github_actions = "github-action"
+    airflow = "airflow"
+
+
+def deploy_command(
+    pipeline_script_path: str,
+    deployment_method: str,
+    schedule: Optional[str],
+    run_on_push: bool,
+    run_on_dispatch: bool,
+    branch: Optional[str] = None
+) -> None:
     # get current repo local folder
     with get_repo(pipeline_script_path) as repo:
-        repo_storage = FileStorage(str(repo.working_dir))
+        deployment_state = DeploymentState(repo, pipeline_script_path, schedule, run_on_push, run_on_dispatch, branch)
+        deployment_state.run_deploy(deployment_method)
+
+
+class DeploymentState:
+    def __init__(
+        self,
+        repo,
+        pipeline_script_path: str,
+        schedule: Optional[str],
+        run_on_push: bool,
+        run_on_dispatch: bool,
+        branch: Optional[str] = None
+    ):
+        self.repo = repo
+        self.pipeline_script_path = pipeline_script_path
+        self.schedule = schedule
+        self.run_on_push = run_on_push
+        self.run_on_dispatch = run_on_dispatch
+        self.branch = branch
+
+        self.repo_storage = FileStorage(str(self.repo.working_dir))
+        self.origin = self._get_origin()
+        # convert to path relative to repo
+        self.repo_pipeline_script_path = self.repo_storage.from_wd_to_relative_path(self.pipeline_script_path)
+        # load a pipeline script and extract full_refresh and pipelines_dir args
+        self.pipeline_script = self.repo_storage.load(self.repo_pipeline_script_path)
+
+        # validate schedule
+        self.schedule_description = self._get_schedule_description()
+        # self.template_storage = utils.clone_command_repo("deploy", branch)
+        self.template_storage = FileStorage("/home/alenaastrakhantseva/dlthub/python-dlt-deploy-template")
```

Comment:
hmmm maybe you can look at `init_command`? this one accepts a local folder as well so it can be passed from command line. and you do not need to hardcode anymore
```
fmt.echo("Looking up the init scripts in %s..." % fmt.bold(repo_location))
    clone_storage = git.get_fresh_repo_files(repo_location, get_dlt_repos_dir(), branch=branch)
```